### DISCUSSION TITLE
Related to #3018

### DISCUSSION THREAD

**Discussion by danisiz** [pods-framework/pods#6631] [Q&A]
Using pods templates I get correct relationship list using {@related_field}, but does not work [each related_field][/each]
At local, i resolve this replacing in file
pods/components/Templates/includes/functions-view_template.php, line 263

```
            if ( isset( $entry[ 'ID' ] ) ) {
                $target_id = $entry[ 'ID' ];            
            } elseif ( isset( $entry[ 'term_id' ] ) ) {
                $target_id = $entry[ 'term_id' ];
            }
```

by:

```
            if ( isset( $entry[ 'ID' ] ) ) {
                $target_id = $entry[ 'ID' ];
            } elseif ( isset( $entry[ 'id' ] ) ) {
                $target_id = $entry[ 'id' ];                    
            } elseif ( isset( $entry[ 'term_id' ] ) ) {
                $target_id = $entry[ 'term_id' ];
            }
```

**Comment by jimtrue** [comment]
Can you please provide
1. What you had in your [each related_field][/each] ie, what was between the each tags that "does not work" and
2. What type of field {@related_field} is (ie what type of relationship and to what type of object?)

**Comment by danisiz** [comment]
I'm working whit an advanced content type, related to another act.
Field in fist act is a relationship to the other act. Config is multiple select, bi-directional field.
In the second act the related field is a relation to the fist one, single select, bi-directional field.

Pod template is like this:

{@name}
{@created} 
{@related} (testing list)

 [if related]
    <table>
        <thead>
        <tr>
                <th>Name</th>
                <th>Field 2</th>  
                <th>Field 3</th>  
            </tr>
        </thead>
        <tbody>

```
[each related]
    \<tr>   
        \<td>{@related.name}</td>
        \<td>{@related.field2}</td>
        \<td>{@related.field3}</td>     
    \</tr>  
[/each]
</tbody>
</table>        
```

[/if]

**Comment by jimtrue** [comment]
It might help if you did a Pods Component: Migrate Packages, Enable and then Pods Migrate, Export (copy that JSON) to a text file and upload so we can reproduce your issue. #3018 was specific to Taxonomy, nothing in your configuration above is specific to Taxonomy. Can you also provide what didn't work? What result did you get?

Within an `[each related]` you don't repeat the {@related.name} within the [each][/each] as you're actually in a related loop at that point. You just use {@name}, {@field2}, etc.

**Comment by danisiz** [comment]
Ok, I tried {@related.name} and {@name} before patching. Now it works well at any of both modes.

Yes, I'm sure, it not works for me ultil patching with first message's code:
Withouth patching, the each block displays every related2 row, not only related ones.

**Comment by jimtrue** [comment]
I'm not doubting you, but we'd be getting reports constantly of issues if this was not working as intended without your code change. That's why having your setup export (which includes your Pods config and your templates and pages), would help for determining if it was something specific with your use case.

**Comment by jimtrue** [comment]
[20160708_sandbox_577fd1927c94c8357160708161514_archive.zip](https://github.com/pods-framework/pods/files/378050/20160708_sandbox_577fd1927c94c8357160708161514_archive.zip)
[export.txt](https://github.com/pods-framework/pods/files/378052/export.txt)

Files were sent to my personal email, so were missed. In the future, please upload them directly to the GitHub ticket so all communication on the issue can stay with the issue.
